pgjdbc/src/main/java/com/yugabyte/ysql/YBjdbcHostnameVerifier.java

```diff
@@ -0,0 +1,225 @@
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+    if(hostlist.contains(hostname))
+      return true;
+    else
+      return false;
+
+
+  }
+
+  protected ArrayList<String> getCurrentServers(Connection conn) throws SQLException {
+    Statement st = conn.createStatement();
+
+    ResultSet rs = st.executeQuery(GET_SERVERS_QUERY);
+    ArrayList<String> currentPrivateIps = new ArrayList<>();
+    String hostConnectedTo = ((PgConnection) conn).getQueryExecutor().getHostSpec().getHost();
+    InetAddress hostConnectedInetAddr;
+
+    boolean isIpv6Addresses = hostConnectedTo.contains(":");
+    if (isIpv6Addresses) {
+      hostConnectedTo = hostConnectedTo.replace("[", "").replace("]", "");
+    }
+
+    try {
+      hostConnectedInetAddr = InetAddress.getByName(hostConnectedTo);
+    } catch (UnknownHostException e) {
+      // This is totally unexpected. As the connection is already created on this host
+      throw new PSQLException(GT.tr("Unexpected UnknownHostException for ${0} ", hostConnectedTo),
+          PSQLState.UNKNOWN_STATE, e);
+    }
+
+    currentPublicIps.clear();
+    while (rs.next()) {
+      String host = rs.getString("host");
+      String publicHost = rs.getString("public_ip");
+      String port = rs.getString("port");
+      String cloud = rs.getString("cloud");
+      String region = rs.getString("region");
+      String zone = rs.getString("zone");
+      hostPortMap.put(host, port);
+      hostPortMapPublic.put(publicHost, port);
+      updateCurrentHostList(currentPrivateIps, host, publicHost, cloud, region, zone);
+      InetAddress hostInetAddr;
+      InetAddress publicHostInetAddr;
+      try {
+        hostInetAddr = InetAddress.getByName(host);
+      } catch (UnknownHostException e) {
+        // set the hostInet to null
+        hostInetAddr = null;
+      }
+      try {
+        publicHostInetAddr = !publicHost.isEmpty()
+            ? InetAddress.getByName(publicHost) : null;
+      } catch (UnknownHostException e) {
+        // set the publicHostInetAddr to null
+        publicHostInetAddr = null;
+      }
+      if (useHostColumn == null) {
+        if (hostConnectedInetAddr.equals(hostInetAddr)) {
+          useHostColumn = Boolean.TRUE;
+        } else if (hostConnectedInetAddr.equals(publicHostInetAddr)) {
+          useHostColumn = Boolean.FALSE;
+        }
+      }
+    }
+    return getPrivateOrPublicServers(currentPrivateIps, currentPublicIps);
+  }
+  protected ArrayList<String> getPrivateOrPublicServers(ArrayList<String> privateHosts,
+      ArrayList<String> publicHosts) {
+    if (useHostColumn == null) {
+      if (publicHosts.isEmpty()) {
+        useHostColumn = Boolean.TRUE;
+      }
+
+      return privateHosts;
+    }
+    ArrayList<String> currentHosts = useHostColumn ? privateHosts : publicHosts
+    return currentHosts;
+  }
+
+  protected void updateCurrentHostList(ArrayList<String> currentPrivateIps, String host,
```

Comment:
This method was added so that `TopologyAwareLoadBalancer` can override it. You can simply replace this method with these two statements in the calling method itself.
```
currentPrivateIps.add(host);
if (!publicHost.trim().isEmpty()) {
    currentPublicIps.add(publicHost);
}
```